index.js

```diff
@@ -51,23 +56,29 @@ const resolve = (source, file, config) => {
  *
  * @param {string} filepath
  * @param {string} packageName
+ * @returns {string} package.json path
  */
 function findPackageJson(filepath, packageName) {
-  for (;;) {
-    const pkgFile = path.join(
-      filepath,
-      "node_modules",
-      packageName,
-      "package.json"
-    );
-    if (fs.existsSync(pkgFile)) {
-      return pkgFile;
-    }
-    const dir = path.dirname(filepath);
-    if (dir === filepath) {
-      return null;
+  try {
+    const resolvedFile = require.resolve(packageName, { paths: [filepath] });
```

Comment:
This will fail when a package only has subpath exports and no main export. E.g. if a package has `my-package/foo` and `my-package/bar` as the only exports, `require.resolve("my-package")` will fail. Can you explain why you think this change is required? Maybe we can find another solution.

Reply:
1. The original function can not find the actual path of the package when the package is a soft link(pnpm), but require.resolve do
2. Global packages which not in the subpath of the folder can not be find.

but these two are not that important.
I will revert it.

Reply:
Did you try this function? `require.resolve.paths`
https://nodejs.org/api/modules.html#requireresolvepathsrequest

seems it can find the resolve path list.